nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/Settings.java

```diff
@@ -81,6 +93,11 @@ public Settings(int defaultConcurrency, int defaultPartitions, Codec<Record> def
         } else {
             this.defaultPolicy = defaultPolicy;
         }
+        if (defaultFilter == null) {
+            this.defaultFilter = RecordFilterChainImpl.NONE;
```

Comment:
You can have a more concise code with Java 9 API:
```java
this.defaultFilter = Objects.requireNonNullElse(defaultFilter, RecordFilterChainImpl.NONE);
```

Reply:
noted, but this code will be backported so stick to java 8